bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2367,7 +2035,324 @@ public String toString () {
 public static Image win32_new(Device device, int type, long handle) {
 	Image image = new Image(device);
 	image.type = type;
-	image.setHandleForZoomLevel(handle, image.getZoom());
+	ImageMetadata imageMetadata = image.new ImageMetadata(handle, image.getZoom());
+	image.setImageMetadataForHandle(imageMetadata, image.getZoom());
```

Comment:
as ImageMetadata is not a static class, can't you move the call image.setImageMetadataForHandle(imageMetadata, image.getZoom()); into the constructor of ImageMetadata? That way we are sure, each time one is instantiated, it will be added to the map in the owning image